### ISSUE TITLE
Error in np.sqrt for large negative values

### ISSUE THREAD

**Issue by Rutzmoser** [numba/numba#3499]
When evaluating the squareroot of a complex number with large negative values, numba produces a wrong number. The minimal example below shows, that the error starts kicking in for numbers starting with 10**-38 and the resulting value is off the exact number by a factor of 0.5. 

## Minimal example
![figure_1](https://user-images.githubusercontent.com/8895501/48503980-f2471e00-e843-11e8-8881-f500bbb13868.png)

```python
import numba
import numpy as np
from matplotlib import pyplot as plt

def squareroot(a):
    return np.sqrt(a)

@numba.jit(nopython=True)
def squareroot_numba(a):
    return np.sqrt(a)

#%%
# create negative complex values
exponents = np.linspace(-10, 100, num=111)
test_vals = - np.array(10**exponents, dtype=complex)

fig, ax = plt.subplots(2, sharex=True)
ax[0].semilogy(exponents, -squareroot(test_vals).imag)
ax[0].plot(exponents, -squareroot_numba(test_vals).imag)
ax[0].grid()
ax[0].set(ylabel='Im(sqrt(-10**x))')

ratio = (squareroot(test_vals) / squareroot_numba(test_vals)).real
ax[1].plot(exponents, ratio, 'x-')
ax[1].set(title='Ratio of sqare roots computed with numpy vs. numba',
          xlabel='exponent to basis 10', ylabel='ratio')
```

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

**Comment by Rutzmoser** [comment]
Sorry, I cannot edit the issue properly in the browser currently available; I'll fix it soon ;-)

**Comment by stuartarchibald** [comment]
Thanks for the report. Seems like you are seeing errors appearing from values `-1e38` and above in magnitude, which is suspiciously near the limit of 32bit float representation. We'll take a look.

**Comment by stuartarchibald** [comment]
My suspicions were correct. The alg was branching to the "scale for overflow" part of the alg based on `FLT_MAX` whereas it should have been type specific (i.e. `DBL_MAX` for complex128``). Fixed in #3527.
